basis/ui/backend/gtk/gtk.factor

```diff
@@ -157,14 +157,39 @@ CONSTANT: events-mask
 : key-sym ( keyval -- string/f action? )
     code>sym [ dup integer? [ gdk_keyval_to_unicode 1string ] when ] dip ;
 
-: key-event>gesture ( event -- key-gesture )
-    [ event-modifiers ] [ keyval>> key-sym ] [
-        type>> GDK_KEY_PRESS = [ <key-down> ] [ <key-up> ] if
-    ] tri ;
-
-: on-key-press/release ( win event user-data -- ? )
-    drop swap [ key-event>gesture ] [ window ] bi* propagate-key-gesture t ;
-
+:: key-event>gesture ( event -- key-gesture )
+    event event-modifiers
+    event keyval>> code>sym drop
+    dup integer? not [
+        drop event keyval>> key-sym
+    ] [
+        gdk_keyval_to_unicode 1string f
+    ] if
+    event type>> GDK_KEY_PRESS = [ <key-down> ] [ <key-up> ] if ;
```

Comment:
You could save a couple of character here if wou wanted to:
```suggestion
    event type>> GDK_KEY_PRESS = <key-down> <key-up> ?;
```

Reply:
That's good.

Reply:
> I’m not sure this suggestion works.

You are right, it does not. Sorry about that...

Reply:
Oh! Surely, it seems better to leave it as "if". Then, I will leave this as it is.